integration_tests/src/main/python/data_gen.py

```diff
@@ -1173,3 +1173,14 @@ def get_25_partitions_df(spark):
         StructField("c3", IntegerType())])
     data = [[i, j, k] for i in range(0, 5) for j in range(0, 5) for k in range(0, 100)]
     return spark.createDataFrame(data, schema)
+
+# If timezone is non-UTC and rebase mode is LEGACY, writing to Parquet will fail because of GPU
+# currently does not support. On Databricks, the default datetime rebase mode is LEGACY,
+# it's different from regular Spark. Some of the cases will fall if timezone is non-UTC on DB.
+# The following configs is for DB and ensure the rebase mode is not LEGACY on DB.
+writer_confs_for_DB = {
```

Comment:
This is a huge change for our testing. I think what we really want for now is to have all of the file read/write operators need to fallback to the CPU if there is a timestamp at all involved. It does not matter if we have LEGACY or not. Until the code has been checked and updated, if needed, to work for other time zones we need to fall back to the CPU and skip the tests like we are doing in other places.

I am not saying that the changes you have done to the plugin are wrong. I am just saying that I don't think this is the right way to fix the tests, and I don't have time right now to fully review that all of the tests you updated didn't lose some kind of coverage that we need/want when this happened.

Reply:
I still don't want this in here. I am fine if we xfail the DB write tests and point to why when the timezone is not UTC. But I don't want to change what we are testing unless we go through each test and verify that we are not losing coverage. This PR is already big enough. If you want to do this change it should be split off into another PR.